src/app/views/query-response/adaptive-cards/AdaptiveCard.tsx

```diff
@@ -21,6 +24,12 @@ class AdaptiveCard extends Component<IAdaptiveCardProps> {
     if (hostConfig) {
       this.adaptiveCard.hostConfig = new AdaptiveCardsAPI.HostConfig(hostConfig);
     }
+    telemetry.trackEvent(TAB_CLICK_EVENT, 
+     {
+       ComponentName: 'Adaptive Cards Tab',
+       QueryUrl: sampleQuery ? sampleQuery.sampleUrl : null,
```

Comment:
Does sampleQuery capture what is in the URL text box? If so, we can't capture this as it can contain information that can identify a user.

Reply:
The best that we can capture is the entity set / singleton being accessed. If we were to do that, I'd like for us to add code to validate that we only capture the entity set / singleton which is the first segment after version. Anything after this segment can be identifiable information.

Reply:
This comment is applicable to all places where we capture query URL.

Reply:
We can get the request url that doesn't have the user information. There is a function that can strip out the rest of the info

Reply:
The function [parseSampleUrl](https://github.com/microsoftgraph/microsoft-graph-explorer-v4/blob/4d9dc4cfbf036f50518210e507fe5762a73450a5/src/app/utils/sample-url-generation.ts#L3) can get the request url and we can leave out the search parameters. However, we can add a new function that can substitute personal information e.g an email, id with a token

Reply:
We need to remove more than just query parameters (I assume that is what you mean by search parameters). We need to remove everything after the entityset/singleton.

https://graph.microsoft.com/v1.0/{entityset|singleton}

Reply:
@MIchaelMainer, I am capturing the query URL to help identify a code snippet or adaptive card, e.g. which are the most popular code snippets.

Reply:
I've excluded query parameters from the URL. Any other PII that may be in the remaining part is being redacted.

Reply:
I've excluded query parameters from the URL. Any other PII that may be in the remaining part is being redacted.